### ISSUE TITLE
Sync Windows malfunction

### ISSUE THREAD

**Issue by alexanderdalloz** [argoproj/argo-cd#5590]
Running Argo CD v1.8.5 and testing the Sync Windows feature I found that creating a deny window blocks syncing even though the system's time is not in the window.

Asking in the slack channel the behaviour got confirmed by jannfis.

```
$ argocd version
argocd: v1.8.5+d0f8edf
  BuildDate: 2021-02-20T05:38:50Z
  GitCommit: d0f8edfec804c013d4fc535e8b9022eb47602617
  GitTreeState: clean
  GoVersion: go1.14.12
  Compiler: gc
  Platform: linux/amd64
argocd-server: v1.8.5+d0f8edf
  BuildDate: 2021-02-20T05:40:24Z
  GitCommit: d0f8edfec804c013d4fc535e8b9022eb47602617
  GitTreeState: clean
  GoVersion: go1.14.12
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: v3.8.1 2020-07-16T00:58:46Z
  Helm Version: v3.4.1+gc4e7485
  Kubectl Version: v1.17.8
  Jsonnet Version: v0.17.0
```
`
$ argocd proj add reference-project-ref --kind deny --schedule "2 13 * * *" --duration 1h --applications "*"`

```
$ argocd proj windows list reference-project-ref
ID  STATUS    KIND  SCHEDULE    DURATION  APPLICATIONS  NAMESPACES  CLUSTERS  MANUALSYNC
0   Inactive  deny  2 13 * * *  1h        *             -           -         Disabled
```

```
$ argocd proj windows list reference-project-ref -o yaml
- applications:
  - '*'
  duration: 1h
  kind: deny
  schedule: 2 13 * * *
```

```
$ argocd app get reference-project-ref-reference-app
Name:               reference-project-ref-reference-app
Project:            reference-project-ref
Server:             https://kubernetes.default.svc
Namespace:          reference-project-ref
URL:                https://argocd-server-argocd.exmple.org/applications/reference-project-ref-reference-app
Repo:               ssh://git@example.org:7999/cdopsrepos/opsrepo_reference-app.git
Target:             stages/ref
Path:               reference-app
SyncWindow:         Sync Allowed
Assigned Windows:   deny:2 13 * * *:1h
Sync Policy:        Automated (Prune)
Sync Status:        Synced to stages/ref (5fc90f2)
Health Status:      Healthy
```

```
$ date -u
Wed Feb 24 09:58:59 UTC 2021
```

```
$ argocd app sync reference-project-ref-reference-app
FATA[0000] rpc error: code = PermissionDenied desc = Cannot sync: Blocked by sync window
```

**Comment by jannfis** [comment]
I could successfully reproduce this behavior also on latest master build. A deny window will block all syncs, also outside the window, even when `--manual-sync` was explicitly specified.

**Comment by jannfis** [comment]
Funny thing is, that when the deny is active and `--manual-sync` is specified to allow manual syncs, a manual sync is actually allowed (ignore the sync errors on non-existing namespace):

```
$ ./dist/argocd --server 127.0.0.1:8080 --plaintext proj windows list default
ID  STATUS  KIND  SCHEDULE    DURATION  APPLICATIONS  NAMESPACES  CLUSTERS  MANUALSYNC
0   Active  deny  2 11 * * *  1h        *             -           -         Enabled
$ ./dist/argocd --server 127.0.0.1:8080 --plaintext app sync guestbook
TIMESTAMP                  GROUP        KIND   NAMESPACE                  NAME      STATUS    HEALTH        HOOK  MESSAGE
2021-02-24T11:06:33+00:00            Service    nonexist  kustomize-guestbook-ui  OutOfSync  Missing              
2021-02-24T11:06:33+00:00   apps  Deployment    nonexist  kustomize-guestbook-ui  OutOfSync  Missing              
2021-02-24T11:06:33+00:00            Service    nonexist  kustomize-guestbook-ui  OutOfSync  Missing              namespaces "nonexist" not found
2021-02-24T11:06:33+00:00   apps  Deployment    nonexist  kustomize-guestbook-ui  OutOfSync  Missing              namespaces "nonexist" not found

Name:               guestbook
Project:            default
...
```
As opposed to an inactive sync window:
```
$ ./dist/argocd --server 127.0.0.1:8080 --plaintext proj windows list default
ID  STATUS    KIND  SCHEDULE    DURATION  APPLICATIONS  NAMESPACES  CLUSTERS  MANUALSYNC
0   Inactive  deny  2 20 * * *  1h        *             -           -         Enabled
$ ./dist/argocd --server 127.0.0.1:8080 --plaintext app sync guestbook
FATA[0000] rpc error: code = PermissionDenied desc = Cannot sync: Blocked by sync window 
```

**Comment by alexanderdalloz** [comment]
The issue still exists in Argo CD v2.0.0

**Comment by hrsikesa** [comment]
Looks like this Deny Windows issue is still active.
I did a workaround & its works fine for me, others can follow same approach for time being this bug is not resolved.
We can create 2 Sync window, 1 Allow Window & other Deny window.
In Allow Window, use these configuration **_schedule_** -> "* * * * *" & _**duration -> 24h**_. Basically enable Sync Allow Window for 24 hrs.
In Deny Window, specify schedule & duration than we want ArgoCD to deny sync.

**Basically, If there is an active matching allow and an active matching deny then syncs will be denied as deny windows override allow windows.**
[Document link](https://argo-cd.readthedocs.io/en/stable/user-guide/sync_windows/)

**Comment by leoluz** [comment]
@hrsikesa https://github.com/argoproj/argo-cd/pull/7517 is supposed to address this issue. The PR is merged but not released yet (its targeted to 2.2). Have you tested with the latest snapshot or with an older version?

**Comment by hrsikesa** [comment]
@leoluz I used this v2.1.5
